subtitles/subtitles.py

```diff
@@ -86,36 +91,33 @@ def serve(properties: dict, debug: bool = False) -> None:
     """
     transcriber = get_transcriber(properties)
     receiver = f'{properties["receiver"]["host"]}:{properties["receiver"]["port"]}'
+    port = properties['api']['port']
 
-    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))  # TODO: How to determine how many workers? Guess?
-    subtitles_pb2_grpc.add_SubtitleGeneratorServicer_to_server(
-        servicer=SubtitleServerService(transcriber, receiver),
-        server=server)
+    with ThreadPoolExecutor(max_workers=10) as executor:  # TODO: How to determine how many workers? Guess?
```

Comment:
First of all the default for this value is most often quite reasonable (`min(32, os.cpu_count() + 4)`).
This allows us to change node size without accidentally leaving this constant in a wrong configuration..

To answer this: how is whispers behaviour?
- Is it CPU/IO/Memory bound?
- How many cores does it use?

Unrelated question: Have you tested, if it releases the GIL and if this is a save operation? (see https://github.com/openai/whisper/discussions/296 why I am confused about this...)

Reply:
The `max_worker` value sounds great. Regarding the other questions I will have to do some more digging and research, but I think you point to an important problem. Big Thanks 💙.

Reply:
There seems to be problems not only with whisper but also with vosk regarding multithreading. As of now it is not possible to run two transcriptions at the same time in the same process but different threads. Which is not so good 🫠

Reply:
Vosk should be just fine.

Reply:
> Vosk should be just fine.

Then it's probably a problem on our side or my tests failed me. 😅 Thanks a lot for the input. ❤️

Reply:
> 

As it seems a simple `mutex` solves the issue.